python/test/test_hail/linalg/test_linalg.py

```diff
@@ -730,3 +730,10 @@ def assert_eq(a, b):
         with self.assertRaises(ValueError) as cm:
             hl.linalg.utils.locus_windows(ht.locus, 1.0, coord_expr=ht.cm)
         self.assertTrue("missing value for 'coord_expr'" in str(cm.exception))
+
+    def test_compute_contig_start_idx(self):
+        res = hl.linalg.utils._compute_contig_start_idx(
+            global_pos=[0, 0, 1, 2, 3, 4, 5, 5],
+            contig_cum_len=[0, 1, 1, 3, 5, 6, 7])
+
+        self.assertEqual(res, [0, 0, 2, 2, 4, 6])
```

Comment:
I'm having a hard time reasoning about this test (also the _compute_contig_start_idx function itself). Maybe it's because you have contig lengths with length == 0? Can you add a test where that doesn't occur?

Reply:
Yeah, I made this robust to the case of contigs of length 0, even if that's unlikely to occur in practice. The value of the start index of each contig is the index of the first value in `global_pos` that is greater than or equal the cumulative length of the preceding contigs. Length 0 contigs at the end, if present, are dropped.